### ISSUE TITLE
Status of org-indent-mode functionality?

### ISSUE THREAD

**Issue by IllustratedMan-code** [nvim-orgmode/orgmode#422]
### Does this feature exist in Emacs orgmode core?

Yes

### Orgmode link

https://orgmode.org/manual/Org-Indent-Mode.html

**Comment by IllustratedMan-code** [comment]
This was talked about before in another issue, but neovim did not support the required functionality. What needs to occur to get this to work? I'm happy to make issues in the main neovim repository to get this to happen.

**Comment by jgollenz** [comment]
Are the virtual spaces just used for non-headline text or is the second space for each headline level also virtual?

```
* foo
  content            <-- two virtual spaces

  * bar              <-- one virtual space?
    more content     <-- four virtual spaces
```

**Comment by IllustratedMan-code** [comment]
without org-indent-mode
```
* foo
content

** bar
more content
```
with org-indent-mode
```
* foo
  content

 ** bar
    more content
```

**Comment by IllustratedMan-code** [comment]
I believe all indents are virtual.

**Comment by jgollenz** [comment]
Ah, it's prepending two virtual spaces to every line, even nested headings.

This 
```
* one
foo
** two
bar
*** three
baz
```
becomes

![image](https://user-images.githubusercontent.com/12050098/194509256-d16817b9-ec2d-4606-8d83-e65049ae541b.png)

**Comment by IllustratedMan-code** [comment]
`org-indent-mode` is also configurable with `org-indent-indentation-per-level` which is my case is set to 2. So the number of virtual spaces prepended should be configurable.

**Comment by IllustratedMan-code** [comment]
Looks like https://github.com/neovim/neovim/pull/9496 needs to be merged before this functionality will be possible.

**Comment by jgollenz** [comment]
@kristijanhusak can you please add the `neovim-dependency` label?

**Comment by danilshvalov** [comment]
See #561

**Comment by treatybreaker** [comment]
The following stills remains to be done:

- [ ] Toggling indent mode on or off in a buffer
- [ ] Support `#+STARTUP: indent` & `#+STARTUP: noindent`

I'll be opening PRs to resolve these as time permits. Probably no later than Sunday `2024-27-01`.
